server/broadcast_test.go

```diff
@@ -323,6 +326,124 @@ func TestSelectSession(t *testing.T) {
 	// XXX check refresh condition more precisely - currently numOrchs / 2
 }
 
+func TestSelectSession_MultipleInFlight(t *testing.T) {
```

Comment:
I see that we're running tests prefixed with `TestSelectSession_` using the race detector, but I don't see any tests involving concurrency below. Concurrent access for the last session seems possible with:

- `selectSession()` called in `transcodeSegment()`
- `completeSession()` called in `transcodeSegment()`
- The update of the `SegsInFlight` for the last session in `transcodeSegment()`

Perhaps we can just setup scenarios where those 3 events occur concurrently (i.e. parallel goroutines).

Reply:
Will send another quick PR for this as this PR blocks other things!